### ISSUE TITLE
Internal error while generating hash with CPasswordHelper::hashPassword()

### ISSUE THREAD

**Issue by NabiKAZ** [yiisoft/yii#3914]
@samdark 

I use PHP 5.3.6 on Windows with Yii 1.1.16
When I use `CPasswordHelper::hashPassword()`, I get this error:
```
CException
Internal error while generating hash.
\framework\utils\CPasswordHelper.php(97)
```
But with Yii 1.1.15 well all done.
Seem you use of `$2y$` instead `$2a$` in `generateSalt()` in 1.1.16 version.
I check this feature in [crypt PHP function](http://php.net/manual/en/function.crypt.php), and see wrote:
>... developers targeting only PHP 5.3.7 and later should use "$2y$" in preference to "$2a$".

Now I have problem with `$2y$` in Yii 1.1.16 on PHP 5.3.6
But there is not any problem with `$2a$` in Yii 1.1.15 version.
Do you thinks is it a bug? :boom:
And so what's your solution for use `CPasswordHelper::hashPassword()` of Yii 1.1.16 on PHP 5.3.6

Now there is a big notice, seen Yii 1.1.16 requirement for `hashPassword()` is PHP **5.3.7** but in the official site wrote: Requirement PHP 5.1.0 or above. Why?!

**Comment by samdark** [comment]
Time to upgrade PHP.

**Comment by NabiKAZ** [comment]
@samdark 
But in the official site wrote: Requirement PHP 5.1.0 or above. Why?!

**Comment by samdark** [comment]
Because overall Yii 1.1 works fine with PHP 5.1 except some classes which are not mandatory for every project.

**Comment by samdark** [comment]
In case of security functionality we prefer to raise requirements instead of allowing building insecure apps.

**Comment by takobell** [comment]
@NabiKAZ Ran into this recently.  It's complicated because the crypt()/blowfish method had a [security](https://secure.php.net/security/crypt_blowfish.php) issue that was officially fixed in PHP 5.3.7. But many linux distributions have back-ported that fix to PHP versions prior to 5.3.7. It's annoying, but given that and not all Yii apps will use CPasswordHelper, I agree with @samdark.

**Comment by NabiKAZ** [comment]
You said correction.
I test it with PHP Version 5.3.3 on CentOS release 6.6, but don't have problem with `$2y$`
So I must search for back-ported fix for my php 5.3.6 on my windows, or upgrade PHP.
